assets/js/components/notifications/EnableAutoUpdateBannerNotification.js

```diff
@@ -0,0 +1,152 @@
+
+const { useSelect, useDispatch } = Data;
+
+const NOTIFICATION_ID = 'enable-plugin-auto-update-notification';
+const HIDE_NOTIFICATION_ON_FIRST_SETUP =
+	'auto-update-banner-hide-notification-on-first-setup';
+
+const EnableAutoUpdateBannerNotification = () => {
+	const hasUpdatePluginCapacity = useSelect( ( select ) =>
+		select( CORE_USER ).hasCapability( PERMISSION_UPDATE_PLUGINS )
+	);
+	const autoUpdatesEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).getAutoUpdatesEnabled()
+	);
+	const siteKitAutoUpdatesEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).getSiteKitAutoUpdatesEnabled()
+	);
+
+	const isDoingEnableAutoUpdate = useSelect( ( select ) =>
+		select( CORE_SITE ).isDoingEnableAutoUpdate()
+	);
+
+	const { enableAutoUpdate } = useDispatch( CORE_SITE );
+
+	const [ notification ] = useQueryArg( 'notification' );
+	const [ slug ] = useQueryArg( 'slug' );
+
+	const [ isInitialPluginSetup, setIsFirstPluginSetup ] = useState( true );
+
+	const setFirstPluginSetup = useCallback(
+		async ( isFirstSetup ) => {
+			if ( isFirstSetup ) {
+				await setItem( HIDE_NOTIFICATION_ON_FIRST_SETUP, true, {
+					ttl: getTimeInSeconds() * 10,
+				} );
+				setIsFirstPluginSetup( isFirstSetup );
+			} else {
+				const { value } = await getItem(
+					HIDE_NOTIFICATION_ON_FIRST_SETUP
+				);
+				setIsFirstPluginSetup( !! value );
+			}
+		},
+		[ setIsFirstPluginSetup ]
+	);
+
+	/**
+	 * If the user just set up Site Kit (eg. just returned from the
+	 * initial OAuth sign-in flow) and is seeing the dashboard
+	 * for the first time, we want to hide the notification for 10
+	 * minutes so they aren't immediately bothered by
+	 * CTA notifications.
+	 */
+	useEffect( () => {
+		if (
+			! hasUpdatePluginCapacity ||
+			! autoUpdatesEnabled ||
+			siteKitAutoUpdatesEnabled
+		) {
+			return;
+		}
+		setFirstPluginSetup(
+			notification === 'authentication_success' && ! slug
+		);
+	}, [
+		notification,
+		slug,
+		hasUpdatePluginCapacity,
+		autoUpdatesEnabled,
+		setFirstPluginSetup,
+		siteKitAutoUpdatesEnabled,
+	] );
+
+	// Don't render anything if the user has no permission to update plugin,
+	// plugin auto-updates are disabled or auto update are already enabled for Site Kit.
+	if (
+		! hasUpdatePluginCapacity ||
+		! autoUpdatesEnabled ||
+		siteKitAutoUpdatesEnabled
+	) {
+		return null;
+	}
+
+	// Don't show this banner if the user just came from the initial Site Kit setup
+	// flow less than 10 minutes ago.
+	if ( isInitialPluginSetup ) {
+		return null;
+	}
+
+	return (
+		<BannerNotification
+			id={ NOTIFICATION_ID }
+			title={ __( 'Keep Site Kit up-to-date', 'google-site-kit' ) }
+			description={ __(
+				'Turn on auto-updates so you always have the latest version of Site Kit. We constantly introduce new features to help you get the insights you need to be successful on the web.',
+				'google-site-kit'
+			) }
+			ctaComponent={
+				<SpinnerButton
+					onClick={ enableAutoUpdate }
```

Comment:
We should add some code here that detects if the request was successful or not and shows an error if it wasn't. Right now if the request fails it's not obvious why:

https://user-images.githubusercontent.com/90871/208753611-afb6c41e-dc1b-4653-9fb3-903a6664abb7.mp4

Reply:
@tofumatt we can not use the existing `nonceEndpoint` ajax action here.
https://github.com/google/site-kit-wp/blob/7f831cbe96e03116a0cc14cae74cea848b680153/includes/Core/Assets/Assets.php#L418-L419

The `rest-nonce` ajax action a feature from WordPress core to generate nonces for reset-requests. https://github.com/WordPress/wordpress-develop/blob/fba12b6eedd73ae41d795d4401e98e4649ceffc9/src/wp-admin/includes/ajax-actions.php#L5448-L5455

This can ajax-action not be used to generate nonces for other actions than `wp_rest`. So we need to implement our own way to generate the nonces. I think we don't want to implement a ajax action for this, as we don't have ajax action in Site Kit.

Do you think there is a better of providing the nonces other than the REST API? Maybe using `inline_base_data`?

https://github.com/google/site-kit-wp/blob/7f831cbe96e03116a0cc14cae74cea848b680153/includes/Core/Assets/Assets.php#L719-L750

Reply:
Okay, if we can't use that nonce (I didn't realise that, I'm actually not super-familiar with how WP handles/manages nonces 😅), I think the approach you have is fine. I don't think we need to move it into `inline_base_data`; it's fine to keep it as-is then 🙂